cni/pkg/repair/netns.go

```diff
@@ -33,46 +31,48 @@ func getPidNamespace(pid int) string {
 	return "/host/proc/" + strconv.Itoa(pid) + "/ns/net"
 }
 
-func runInHost[T any](f func() (T, error)) (T, error) {
+func runInNS[T any](pidNs string, f func() (T, error)) (T, error) {
 	var res T
-	ns, err := netns.GetNS(getPidNamespace(1))
+	ns, err := netns.GetNS(pidNs)
 	if err != nil {
-		return res, fmt.Errorf("failed to get host network: %v", err)
+		return res, fmt.Errorf("failed to get network namespace %v: %v", pidNs, err)
 	}
 	err = ns.Do(func(ns netns.NetNS) error {
 		var err error
 		res, err = f()
 		return err
 	})
 	if err != nil {
-		return res, fmt.Errorf("in host network: %v", err)
+		return res, fmt.Errorf("in network namespace %v: %v", ns, err)
 	}
 	return res, nil
 }
 
-func findNetworkIDByIP(ip string) (int, error) {
-	link, err := getLinkWithDestinationOf(ip)
-	if err != nil {
-		return 0, fmt.Errorf("find link for %v: %v", ip, err)
-	}
-	return link.Attrs().NetNsID, nil
+func runInHost[T any](f func() (T, error)) (T, error) {
+	return runInNS(getPidNamespace(1), f)
```

Comment:
nit: Under what contexts would this fail? Would it work if invoked from inside a container? Or does it need to be called _from_ the host node - might be worth noting that in a comment if so.